Zephyr USB TCPC Driver API and Sample

sambhurst opened the pull request:
These pull requests are part of a larger effort to port the Google ChromeOS USB-C Stack to Zephyr. See issue #38371.
This pull requests contains the USB TCPC Driver API, STM32 TCPC Driver, and Sample application based on the STM32 TCPC Driver.

sambhurst committed:
usbc: Add USB TCPC Driver API and support PD structures

This commit adds the API for USB Type-C Port Controllers (TCPC) and support
Power Delivery (PD) structures

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
stm32: Add STM32 USB TCPC Driver

This commit adds a USB Type-C Port Controller Driver for
the STM32 USB Type-C / USB Power Delivery (UCPD) peripheral

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
samples: boards: Add sample application based on the STM32 USB TCPC Driver

This sample application uses the STM32 USB TCPC Driver to create a USBC
Sink Application for the STM32G081b_eval board.

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

gmarull commented on include/drivers/usbc/usbc_tc.h:
```diff
@@ -0,0 +1,435 @@
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
```
nit: this header doesn't seem to use any of the device APIs

sambhurst replied:
Done

gmarull commented on include/drivers/usbc/usbc_tcpc.h:
```diff
@@ -0,0 +1,1020 @@
+	const struct tcpc_driver_api *api =
+		(const struct tcpc_driver_api *)dev->api;
+
+	if (api->set_vconn == NULL) {
+		return -ENOTSUP;
+	}
```
question: shouldn't that be `-ENOSYS`?

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/Kconfig.tcpc_stm32:
```diff
@@ -0,0 +1,13 @@
+
+config USBC_TCPC_STM32
+	bool "USBC TCPC device controller driver for STM32 devices"
+	depends on SOC_FAMILY_STM32
```
nit: may be removed (line below implies this)

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/Kconfig.tcpc_stm32:
```diff
@@ -0,0 +1,13 @@
+	bool "USBC TCPC device controller driver for STM32 devices"
+	depends on SOC_FAMILY_STM32
+	depends on (SOC_SERIES_STM32G0X || SOC_SERIES_STM32G4X || SOC_SERIES_STM32L5X || SOC_SERIES_STM32U5X)
+	select USE_STM32_LL_UCPD
```
suggestion: default based on DT status `default $(dt_compat_enabled, ...)`

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+		uint32_t cr = LL_UCPD_ReadReg(config->ucpd_port, CR);
+		int pol = !!(cr & UCPD_CR_PHYCCSEL);
+
+		mask &= ~(1 << (UCPD_CR_CCENABLE_Pos + !pol));
```
`BIT(...)`

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+/*
```
nit: filename uses `g`, but seems to apply to `L5/U5`

sambhurst replied:
I renamed the file stm32_ucpd.c

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+ * @return true if message is pending, else false
+ * @return -EIO on failure
+ */
+static bool ucpd_is_rx_pending_msg(const struct device *dev,
+				   enum pd_packet_type *type)
+{
+	struct tcpc_data *data = dev->data;
+	int ret = 0;
+
+	ret = (*(uint32_t *)data->ucpd_rx_buffer > 0);
+
+	if (ret & (type != NULL)) {
+		*type = *(uint16_t *)data->ucpd_rx_buffer;
+	}
+
+	return ret;
```
function returls bool, but `int` is used here

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+	struct tcpc_data *data = dev->data;
+	int ret = 0;
+
+	if (!msg) {
```
nit: better use essentially boolean expressions: `(msg != NULL)`

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+ * @brief This macro converts the psc_ucpdclk pre-scaler division value
+ *        to a register value. reg_val = log_base_2(v)
+ */
+#define PSC_UCPDCLK_MAP(v) \
```
would be nice to introduce a generic log2 function, ref. https://elixir.bootlin.com/linux/latest/source/include/linux/log2.h

sambhurst replied:
I used include/math/ilog2.h

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+			      &drv_data_##inst,					\
+			      &drv_config_##inst,				\
+			      POST_KERNEL,					\
+			      CONFIG_KERNEL_INIT_PRIORITY_DEVICE,		\
```
please, introduce a driver type init priority, e.g. `CONFIG_USBC_INIT_PRIORITY`.

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx.c:
```diff
@@ -0,0 +1,1442 @@
+#define TCPC_DRIVER_INIT(inst)							\
+	static struct tcpc_data drv_data_##inst;				\
+	static const struct tcpc_config drv_config_##inst = {			\
+		.ucpd_port =							\
+			(UCPD_TypeDef *)DT_INST_REG_ADDR(inst),			\
```
nit: fits into 1 single line

sambhurst replied:
Done

gmarull commented on drivers/usbc/tcpc/ucpd_stm32gx_priv.h:
```diff
@@ -0,0 +1,309 @@
+#ifndef ZEPHYR_DRIVERS_USBC_DEVICE_UCPD_STM32X_PRIV_H_
+#define ZEPHYR_DRIVERS_USBC_DEVICE_UCPD_STM32X_PRIV_H_
+
+#include <drivers/usbc/usbc_tcpc.h>
+#include <stm32_ll_ucpd.h>
```
nit: headers should be self-contained: e.g. misses `sys/util_macro.h`

sambhurst replied:
I added #include <sys/util.h>

gmarull commented on samples/drivers/usbc/sink/boards/stm32g071b_disco.overlay:
```diff
@@ -0,0 +1,88 @@
+	hbitclkdiv = <27>;
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+	status = "okay";
```
if board always uses these pins for I2C, this part should be moved there.

sambhurst replied:
Done

gmarull commented on samples/drivers/usbc/sink/prj_stm32g071b_disco.conf:
```diff
@@ -0,0 +1,10 @@
+CONFIG_LOG=y
+
+CONFIG_SENSOR=y
+CONFIG_GPIO=y
```
boards usually have GPIO on by default

sambhurst replied:
Done

gmarull commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,130 @@
+{
+	int ret;
+
+	vbus = device_get_binding(VBUS_LABEL);
+	if (!device_is_ready(vbus)) {
+		LOG_ERR("VBUS device not ready\n");
+		return -EIO;
+	}
+
+	led = device_get_binding(LED_LABEL);
```
please use `DEVICE_DT_GET` everywhere

sambhurst replied:
Done

gmarull commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,130 @@
+		return -EIO;
+	}
+
+	ret = gpio_pin_configure(rdcc1, RDCC1_PIN, GPIO_OUTPUT_ACTIVE | RDCC1_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure(encc1, ENCC1_PIN, GPIO_OUTPUT_ACTIVE | ENCC1_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure(encc2, ENCC2_PIN, GPIO_OUTPUT_ACTIVE | ENCC2_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
+	gpio_pin_set(rdcc1, RDCC1_PIN, 1);
+	gpio_pin_set(encc1, ENCC1_PIN, 1);
+	gpio_pin_set(encc2, ENCC2_PIN, 0);
+
+	ret = gpio_pin_configure(led, LED_PIN, GPIO_OUTPUT_ACTIVE | LED_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
```
nit: use `gpio_dt_spec`

sambhurst replied:
Done

gmarull reviewed (commented):
I haven't reviewed the implementation in detail, just have a few comments.

sambhurst committed:
boards: arm: stm32g071b_disco: Add I2C device tree node

Add I2C node to board level device tree and ina230 I2C device

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

gmarull commented on samples/drivers/usbc/sink/boards/stm32g081b_eval.overlay:
```diff
@@ -0,0 +1,66 @@
+	 * designed to work in freq ranges of 6 <--> 18 MHz, however recommended
+	 * range is 9 <--> 18 MHz.
+	 *
+	 *          ------- @ 16 MHz ---------   @ ~600 kHz   -------------
+	 * HSI ---->| /psc |-------->| /hbit |--------------->| trans_cnt |
+	 *          -------          ---------    |           -------------
+	 *                                        |           -------------
+	 *                                        |---------->| ifrgap_cnt|
+	 *                                                    -------------
```
```suggestion
	 *          +------+ @ 16 MHz +-------+   @ ~600 kHz   +-----------+
	 * HSI ---->| /psc |--------->| /hbit |--------------->| trans_cnt |
	 *          +------+          +-------+    |           +-----------+
	 *                                         |           +-----------+
	 *                                         +---------->| ifrgap_cnt|
	 *                                                     +-----------+
```

sambhurst replied:
Done

gmarull commented on samples/drivers/usbc/sink/CMakeLists.txt:
```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
```
note: sample misses `sample.yaml`?

sambhurst replied:
Done

gmarull commented on include/drivers/usbc/usbc_tcpc.h:
```diff
@@ -0,0 +1,1020 @@
+ * @brief TCPC Chip Information
+ */
+struct tcpc_chip_info {
+	/* Vendor Id */
+	uint16_t vendor_id;
+	/* Product Id */
```
Please use Doxygen docstrings everywhere:

```suggestion
	/** Vendor Id */
	uint16_t vendor_id;
	/** Product Id */
```

sambhurst replied:
Done

gmarull commented on include/drivers/usbc/usbc_tcpc.h:
```diff
@@ -0,0 +1,1020 @@
+#ifndef ZEPHYR_INCLUDE_DRIVERS_USBC_USBC_TCPC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_USBC_USBC_TCPC_H_
+
+/**
+ * @brief USB Type-C Port Controller API
+ * @defgroup _usb_type_c_port_controller_api USB Type-C Port Controller API
+ * @{
+ */
```
This defines a top group, however, all drivers now go into `io_interfaces` (check others)

sambhurst committed:
boards: arm: stm32g081b_eval: Add ADC1 device tree node

Add ADC1 node to board level device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
boards: arm stm32g081b_eval: Add Voltage-Divider device tree node

Add Voltage-Divider node for VBUS to board level device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
boards: arm: Add "tcpc" feature to yaml supported

Add "tcpc" feature to yaml supported property for boards
stm32g071b_disco and stm32g081b_eval

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+
+#define TCPC_NODE       DT_ALIAS(tcpc)
+
+#if DT_NODE_HAS_STATUS(TCPC_NODE, okay)
+#define TCPC_DEVICE     DT_LABEL(TCPC_NODE)
+#else
+/* A build error here means your board isn't set up to build this sample. */
+#error "Unsupported board: tcpc devicetree alias is not defined"
+#define TCPC_DEVICE ""
+#endif
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+	tc.tcpc_dev = DEVICE_DT_GET(TCPC_NODE);
+	if (!device_is_ready(tc.tcpc_dev)) {
+		LOG_ERR("TCPC device not ready\n");
+		return -EIO;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+		tc->flag |= FLAG_PDMSG_SENT;
+		break;
+	/* These alerts are ignored */
+	case TCPC_ALERT_VBUS_ALARM_HI:
+	case TCPC_ALERT_VBUS_ALARM_LO:
+	case TCPC_ALERT_FAULT_STATUS:
+	case TCPC_ALERT_RX_BUFFER_OVERFLOW:
+	case TCPC_ALERT_VBUS_SNK_DISCONNECT:
+	case TCPC_ALERT_BEGINNING_MSG_STATUS:
+	case TCPC_ALERT_EXTENDED_STATUS:
+	case TCPC_ALERT_EXTENDED:
+	case TCPC_ALERT_VENDOR_DEFINED:
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+static void alert_handler(const struct device *dev,
+			  void *data, enum tcpc_alert alert)
+{
+	struct tc_t *tc = (struct tc_t *)data;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+		/* Display Soruce Caps Received form Source */
+		display_source_caps(tc);
+		tc->pd_state = SNK_READY;
+		printk("SNK_READY\n");
+	/* fall through */
+	case SNK_READY:
+		/* STAY HERE FOREVER */
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+				/* Waiting for message to send */
+				break;
+			}
+		/* fall through */
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+
+		/* Move on to the select capabilities */
+		tc->pd_state = SNK_SELECT_CAPABILITY;
+		tc->sub_state = 0;
+	/* Fall through */
+	case SNK_SELECT_CAPABILITY:
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+			break;
+		}
+		/* We got the source caps message */
+		tc->pd_state = SNK_EVALUATE_CAPABILITY;
+	/* Fall through */
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+	/* Fall through */
+	case SNK_SELECT_CAPABILITY:
+		switch (tc->sub_state) {
+		case 0:
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+
+static void tc_unattached_snk_run(void *tc_obj)
+{
+	struct tc_t *tc = (struct tc_t *)tc_obj;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+ */
+static void tc_unattached_snk_entry(void *tc_obj)
+{
+	struct tc_t *tc = (struct tc_t *)tc_obj;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,730 @@
+		/* Update timer */
+		tc.cc_debounce--;
+
+		/* Did timer expier? */
+		if (tc.cc_debounce) {
+			return;
+		}
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+	dev_adc = DEVICE_DT_GET(ADC_NODE);
+	if (!device_is_ready(dev_adc)) {
+		printk("ADC device not found\n");
+		return -EIO;
+	}
+
+	if (!device_is_ready(led.port)) {
+		LOG_ERR("LED device not ready");
+		return -EIO;
+	}
+
+	if (!device_is_ready(discharge_vbus.port)) {
+		LOG_ERR("Discharge VBUS device not ready");
+		return -EIO;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+
+void board_led(int val)
+{
+	gpio_pin_set(led.port, led.pin, val);
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+ */
+int board_discharge_vbus(bool en)
+{
+	gpio_pin_set(discharge_vbus.port, discharge_vbus.pin, (int)en);
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+/**
+ * @brief This function discharges VBUS
+ */
+int board_discharge_vbus(bool en)
```
Change to int

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+	ret = adc_read(dev_adc, &sequence);
+	if (ret != 0) {
+		printk("ADC reading failed with error %d.\n", ret);
+		return 0;
```
Changed the interface to int board_vbus_meas(int *mv) and return error codes

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+	ret = adc_raw_to_millivolts(ADC_REF_MV, ADC_GAIN, ADC_RESOLUTION, &value);
+	if (ret) {
+		printk("Scaling ADC failed with error %d.\n", ret);
+		return 0;
```
Changed the interface to int board_vbus_meas(int *mv) and return error codes

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+		return 0;
+	}
+
+	/*
+	 * VBUS is scaled down though a voltage divider with,
+	 * where R1 = 330K and R2 = 49.9K.
+	 *
+	 *            R2       49900
+	 * Scale =  ------- =  ------ = 0.131
+	 *          R1 + R2    379900
+	 */
+
+	return (value * 1000) / 131;
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g081b_eval_board.c:
```diff
@@ -0,0 +1,135 @@
+#define DISCHARGE_VBUS_NODE    DT_PATH(vbus_config, discharge_vbus2)
+#define LED0_NODE              DT_ALIAS(led0)
+
+#define DISCHARGE_VBUS_FLAGS   DT_GPIO_FLAGS(DISCHARGE_VBUS_NODE, gpios)
+#define LED_FLAGS              DT_GPIO_FLAGS(LED0_NODE, gpios)
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,108 @@
+	if (ret < 0) {
+		return ret;
+	}
+	gpio_pin_set(rdcc1.port, rdcc1.pin, 1);
+	gpio_pin_set(encc1.port, encc1.pin, 1);
+	gpio_pin_set(encc2.port, encc2.pin, 0);
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,108 @@
+		return -EIO;
+	}
+
+	ret = gpio_pin_configure_dt(&rdcc1, GPIO_OUTPUT_ACTIVE | RDCC1_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&encc1, GPIO_OUTPUT_ACTIVE | ENCC1_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&encc2, GPIO_OUTPUT_ACTIVE | ENCC2_FLAGS);
+	if (ret < 0) {
+		return ret;
+	}
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,108 @@
+	int ret;
+
+	vbus = DEVICE_DT_GET(VBUS_NODE);
+	if (!device_is_ready(vbus)) {
+		LOG_ERR("VBUS device not ready\n");
+		return -EIO;
+	}
+
+	if (!device_is_ready(rdcc1.port)) {
+		LOG_ERR("RDCC1 device not ready");
+		return -EIO;
+	}
+
+	if (!device_is_ready(encc1.port)) {
+		LOG_ERR("ENCC1 device not ready");
+		return -EIO;
+	}
+
+	if (!device_is_ready(encc2.port)) {
+		LOG_ERR("ENCC2 device not ready");
+		return -EIO;
+	}
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,108 @@
+
+void board_led(int val)
+{
+	gpio_pin_set(led.port, led.pin, val);
```
Done

sambhurst commented on samples/drivers/usbc/sink/src/stm32g071b_disco_board.c:
```diff
@@ -0,0 +1,108 @@
+#define ENCC1_NODE      DT_PATH(cc_config, encc1)
+#define ENCC2_NODE      DT_PATH(cc_config, encc2)
+
+#define LED_FLAGS       DT_GPIO_FLAGS(LED0_NODE, gpios)
+#define RDCC1_FLAGS     DT_GPIO_FLAGS(RDCC1_NODE, gpios)
+#define ENCC1_FLAGS     DT_GPIO_FLAGS(ENCC1_NODE, gpios)
+#define ENCC2_FLAGS     DT_GPIO_FLAGS(ENCC2_NODE, gpios)
```
Done

sambhurst commented on samples/drivers/usbc/sink/boards/stm32g081b_eval.overlay:
```diff
@@ -0,0 +1,66 @@
+ * ADC1, input 3, is used to measure VBUS
+ */
+&adc1 {
+	pinctrl-0 = <&adc1_in3_pa3>;
+        pinctrl-names = "default";
```
Moved this to board level device tree

erwango commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+	int ret;
+
+	/* The UCPD port is disabled in the LL_UCPD_Init function */
+	ret = LL_UCPD_Init(config->ucpd_port,
```
Just a comment: By using this function, you spare the need of defining in dts and setting clock using zephyr clock API.
This is not blocking, but having clocks from dts and using clock api would help in doing PM operation in future.

sambhurst replied:
I added a note that reflects your comment.

erwango reviewed (changes requested):
Thanks for this! A bunch of comments and suggestions, but nothing blocking.

sambhurst committed:
boards: arm: stm32g071b_disco: Add UCPD1 device tree node

Add UCPD1 node to board level device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
boards: arm: stm32g071b_disco: Add CC Confiuration lines

Add CC configuration lines to board level device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
boards: arm: stm32g081b_eval: Add UCPD2 device tree node

Add UCPD2 node to board level device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst committed:
boards: arm: stm32g081b_eval: Add VBUS2 Discharge configuration

Add VBUS2 Discharge configuration node to device tree

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst commented on drivers/usbc/tcpc/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_USBC_TCPC_STM32 stm32_ucpd.c)
```
Done

sambhurst commented on drivers/usbc/tcpc/Kconfig.tcpc_stm32:
```diff
@@ -0,0 +1,15 @@
+
+config USBC_TCPC_STM32
+	bool "USBC TCPC device controller driver for STM32 devices"
+	depends on (SOC_SERIES_STM32G0X || SOC_SERIES_STM32G4X || SOC_SERIES_STM32L5X || SOC_SERIES_STM32U5X)
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+#define DT_DRV_COMPAT st_stm32_ucpd
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ucpd_stm32gx, CONFIG_USBC_LOG_LEVEL);
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+	/* SOP'/SOP'' must be enabled via TCPCI call */
+	data->ucpd_rx_sop_prime_enabled = false;
+
+	stm32gx_ucpd_state_init(dev);
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+
+		IRQ_CONNECT(DT_INST_IRQN(0),
+			    DT_INST_IRQ(0, priority),
+			    ucpd_isr, dev_inst, 0);
```
This code passes a pointer to an array of device instances to the isr:
void ucpd_isr(const struct device *dev_inst[])

erwango replied:
Ha, right. Thanks

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+				UCPD_SR_HRSTSENT |
+				UCPD_SR_HRSTDISC;
+
+	if (IS_ENABLED(CONFIG_SOC_SERIES_STM32G0X)) {
```
This G0 series specific code is guarded by the IS_ENABLE(CONFIG_SOC_SERIES_STM32G0X) line.

erwango replied:
Indeed, but the code guarded es only removed at linkage. So, even though the code is guarded, you'll get errors at build time if symbols are unknown to the target (which is the case here for `SYSCFG_ITLINE8_SR_UCPD1_Msk` on non G0).
You can have a try with, for instance:
```
	if (IS_ENABLED(CONFIG_SOC_SERIES_STM32G4X)) {
		unkown ++;
	}
```

sambhurst replied:
I changed the IS_ENABLED to #ifdef

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+		 * Since the UCPD peripherals share the same interrupt line, determine
+		 * which one generated the interrupt.
+		 */
+		if (SYSCFG->IT_LINE_SR[8] & SYSCFG_ITLINE8_SR_UCPD1_Msk) {
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+		if (SYSCFG->IT_LINE_SR[8] & SYSCFG_ITLINE8_SR_UCPD1_Msk) {
+			/* UCPD1 interrupt is pending */
+			ucpd_base = UCPD1_BASE;
+		} else if (SYSCFG->IT_LINE_SR[8] & SYSCFG_ITLINE8_SR_UCPD2_Msk) {
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+ * @brief Zephyr has difficulty with shared interrupt. This flag is used to
+ *        configure a shared interrupt once and is set to false afterwards.
+ */
+static bool irq_config = true;
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+static void update_stm32g0x_cc_line(UCPD_TypeDef *ucpd_port)
+{
+	if ((uint32_t)(ucpd_port) == UCPD1_BASE) {
+		SYSCFG->CFGR1 |= SYSCFG_CFGR1_UCPD1_STROBE_Msk;
+	} else {
+		SYSCFG->CFGR1 |= SYSCFG_CFGR1_UCPD2_STROBE_Msk;
```
This function is compiled out for other series because its use is guarded:

if (IS_ENABLED(CONFIG_SOC_SERIES_STM32G0X)) {
                update_stm32g0x_cc_line(config->ucpd_port);
}

erwango replied:
cf feedback on previous remark

sambhurst replied:
I changed the IS_ENABLED to #ifdef

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+/**
+ * @brief Initializes the RX and TX state machine variables
+ */
+static void stm32gx_ucpd_state_init(const struct device *dev)
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+	const struct tcpc_config *const config = dev->config;
+
+	if (enable) {
+		LL_UCPD_Enable(config->ucpd_port);
+	} else {
+		LL_UCPD_Disable(config->ucpd_port);
```
Done

sambhurst commented on boards/arm/stm32g071b_disco/stm32g071b_disco.dts:
```diff
@@ -104,6 +104,13 @@
 };
 
+&i2c1 {
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
```
Done

sambhurst commented on boards/arm/stm32g081b_eval/stm32g081b_eval.dts:
```diff
@@ -104,6 +104,12 @@
 };
 
+&adc1 {
+	pinctrl-0 = <&adc1_in3_pa3>;
```
Done

sambhurst commented on drivers/usbc/tcpc/Kconfig:
```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+config USBC_TCPC_DRIVER
```
I copied the convention from usb/device/Kconfig.
I'll remove it.

sambhurst commented on samples/drivers/usbc/sink/boards/stm32g071b_disco.overlay:
```diff
@@ -0,0 +1,84 @@
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
```
Done

sambhurst commented on samples/drivers/usbc/sink/boards/stm32g081b_eval.overlay:
```diff
@@ -0,0 +1,53 @@
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
```
Done

sambhurst commented on samples/drivers/usbc/sink/prj_stm32g071b_disco.conf:
```diff
@@ -0,0 +1,9 @@
+CONFIG_USBC_TCPC_DRIVER=y
+CONFIG_USBC_TCPC_STM32=y
+CONFIG_SMF=y
+CONFIG_LOG=y
```
Done

sambhurst commented on drivers/usbc/tcpc/stm32_ucpd.c:
```diff
@@ -0,0 +1,1426 @@
+{
+	struct tcpc_data *data = dev->data;
+
+	/* Update static varialbe used to filter SOP//SOP'' messages */
```
Done

sambhurst commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1422 @@
+
+		/* Initialize the isr */
+		ucpd_isr_init(dev);
+	}
```
Done

sambhurst commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1422 @@
+		header.message_type == PD_CTRL_GOOD_CRC);
+}
+
+#ifdef CONFIG_SOC_SERIES_STM32G0X
```
This was a requested change by @erwango

erwango replied:
cf https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r787637845.

brockus-zephyr replied:
Got it... the reason is a defined symbol in this code is only defined if CONFIG_SOC_SERIES_STM32G0X is defined and will give a compile error if it is not... Thanks

sambhurst commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1422 @@
+	/* Apply cc pull resistor change */
+	LL_UCPD_WriteReg(config->ucpd_port, CR, cr);
+
+#ifdef CONFIG_SOC_SERIES_STM32G0X
```
This was a requested change by @erwango

sambhurst commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1422 @@
+	/* Update pull values */
+	LL_UCPD_WriteReg(config->ucpd_port, CR, cr);
+
+#ifdef CONFIG_SOC_SERIES_STM32G0X
```
This was a requested change by @erwango

sambhurst commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1422 @@
+
+	if (data->ucpd_vconn_enable) {
+		uint32_t cr = LL_UCPD_ReadReg(config->ucpd_port, CR);
+		int pol = !!(cr & UCPD_CR_PHYCCSEL);
```
Good catch

sambhurst commented:
Hello @erwango, are there any other recommended changes you have?

tejlmand commented on samples/drivers/usbc/sink/CMakeLists.txt:
```diff
@@ -0,0 +1,14 @@
+target_sources(app PRIVATE src/usbc_snk.c)
+
+if("${BOARD}" STREQUAL "stm32g071b_disco")
+  target_sources(app PRIVATE src/stm32g071b_disco_board.c)
+else()
+  target_sources(app PRIVATE src/stm32g081b_eval_board.c)
```
those seems to be very board specific sources.

Generally we should avoid such constructs, especially because it makes it harder to reuse a sample on another board which shares same SoC or peripheral as those boards.

Is it not possible to select source file based on devicetree knowledge or Kconfig setting ?

And if not, what makes this sample so special it should be restricted to those board, and not a general restriction based on hw capabilities.
For example, like the bluetooth samples of course requires a radio peripheral, but we don't test for `n` different boards in those samples.

At this point i'm asking in order to get a better understanding.

sambhurst replied:
Yes, those files are very board specific. They implement a common API used by the sample to configure the board for USBC operation. For example, the stm32g081b eval board uses on on chip ADC to measure VBUS but the stm32g071b disco board uses an external I2C INA device to measure VBUS.

tejlmand replied:
> stm32g071b disco board uses an external I2C INA device to measure VBUS

with this, I assume you refer to the `ti,ina230` compatible device: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/sensor/ti%2Cina230.yaml

That would mean that any board / shield having the `ti,ina230` compatible should be working, correct ?

Then that should be your test, and not the board.

sambhurst replied:
Sure, but some boards might measure VBUS in some other way other than ti,ina230 or an internal ADC. Also, this was just one example, there could be several GPIOs that need configured to properly enable USBC functionality.

tejlmand replied:
>  there could be several GPIOs that need configured to properly enable USBC functionality.

Sure, but that's still not a reason to be board centric.
Check for requirements that are needed for the sample.

Just like build the can driver sample:
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/can

when building for a board without can, for example nrf52840dk_nrf52840:
```
/projects/github/ncs/zephyr/include/device.h:96:39: error: '__device_dts_ord_DT_CHOSEN_zephyr_canbus_ORD' undeclared here (not in a function)
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
```

For the TCPC sample, there could even be someone adding a shield so that the sample could work on even more boards.

If this really is a board specific sample (which I don't think from the look of it), then it ought to go to https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/boards/ instead.

sambhurst replied:
I'm sorry, but I still think I'm missing something.

In the sample, main calls the board_config() function and for:
stm32g081b_eval: board_config() configures the ADC, and a pin for VBUS discharge
stm32g071b_disco: board_config() configures three pins that enable control of the CC lines

The two specific board_config() function do entirely different things that are need for the sample to work correctly.

I'm not sure how to do that through the device tree.

Also, since the board_config() functions configure things for this sample only, I don't think they should be moved to the board directory.

erwango replied:
What about introducing a dedicated choice Kconfig symbol  for this sample:
USBC_CONFIGURE_VBUS
USBC_CONFIGURE_ADC
There are numerous samples that add a Kconfig file for their specific configuration needs, look at for instance  `samples/subsys/usb/mass/Kconfig`

sambhurst replied:
I moved the sample to the boards directory and limited it to the stm32g081b_eval board.

sambhurst replied:
@tejlmand when you have time, could you take another look at my changes. Thank you.

tejlmand reviewed (changes requested):
some minor nits and an observation.

sambhurst commented on drivers/usbc/tcpc/Kconfig:
```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config USBC_TCPC
+	bool "Enable USBC TCPC drivers"
```
Done

sambhurst commented on drivers/usbc/tcpc/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_USBC_TCPC)
```
Done

sambhurst commented on drivers/usbc/tcpc/Kconfig.tcpc_stm32:
```diff
@@ -0,0 +1,15 @@
+
+config USBC_TCPC_STM32
+	bool "USBC TCPC device controller driver for STM32 devices"
+	depends on SOC_FAMILY_STM32
```
Done

mbolivar-nordic reviewed (changes requested):
This has the same missing documentation problem as bbram: https://github.com/zephyrproject-rtos/zephyr/issues/42842

sambhurst committed:
doc: drivers: tcpc: Add USB TCPC API documentation

This commit adds documentation for the
USB Type-C Port Controller (TCPC) APIs.

Signed-off-by: Sam Hurst <sbh1187@gmail.com>

sambhurst commented:
> This has the same missing documentation problem as bbram: #42842

@mbolivar-nordic Thanks for pointing this out. I've added the documentation. Please take a look at your earliest convenience.

mbolivar-nordic commented on doc/reference/api/overview.rst:
```diff
@@ -287,6 +287,11 @@ current :ref:`stability level <api_lifecycle>`.
 
+   * - :ref:`tcpc_api`
+     - Experimental
+     - 3.0
+     - 3.0
```
I doubt this is going to make it for v3.0; did you mean 3.1 instead here?

jfischer-no commented on include/drivers/usbc/usbc_tcpc.h:
```diff
@@ -0,0 +1,1067 @@
+					   tcpc_vbus_cb_t vbus_cb);
+typedef void (*tcpc_set_discharge_vbus_cb_t)(const struct device *dev,
+					     tcpc_discharge_vbus_cb_t discharge_vbus_cb);
+typedef bool (*tcpc_check_vbus_level_t)(const struct device *dev,
```
I guess all of these not `*_cb_t `typedefs are not necessary.

sambhurst replied:
I thought using typedefs would make it easier to read the struct tcpc_driver_api {

jfischer-no replied:
I disagree. Besides, these are only used in this header fine, so they are not necessary.

jfischer-no commented on include/drivers/usbc/usbc_tc.h:
```diff
@@ -0,0 +1,435 @@
+/* usbc_tc.h - USB Type-C Cable and Connector */
```
redundant, please remove this line

sambhurst replied:
Done

jfischer-no commented on samples/boards/stm32/usbc/sink/CMakeLists.txt:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
```
Why is it `samples/boards/stm32/usbc ` and not `samples/subsys/usbc`? Does it not use generic USBC API?

sambhurst replied:
I initially had it in  samples/subsys/usbc but it was recommended that I move it to samples/boards/stm32/usbc because dev-board specific files were needed to configure the board for USBC operation.

jfischer-no replied:
It would be better to have this sample in `samples/subsys/usbc`. One might think the USBC driver API is not generic enough. Besides, in the sample itself there is not much board specific. [stm32g081b_eval_board.c](https://github.com/zephyrproject-rtos/zephyr/pull/41856/commits/2ab55ab8928652510ce867d8f357925523b40a12#diff-600bd7c0d7922f99cba8131489cbdc91723bda56fbc258c1e6db7eb6c8e8e8c5) uses generic ADC driver API as I see and could be abstracted further.

sambhurst replied:
I initially had it there but it was recommended that I move it to boards. I'm inclined to leave this sample here because I have another PR that implements the full USBC subsystem stack that included a more general sample.

tejlmand replied:
> but it was recommended that I move it to boards.

No, it was recommended that this sample was not board centric in its code, but instead be made generic.
https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r796752627
> Generally we should avoid such constructs, especially because it makes it harder to reuse a sample on another board which shares same SoC or peripheral as those boards.
>
> Is it not possible to select source file based on devicetree knowledge or Kconfig setting ?
>
> And if not, what makes this sample so special it should be restricted to those board, and not a general restriction based on hw capabilities.

and then later:
https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r797578264
> Sure, but that's still not a reason to be board centric.


Only as an alternative solution to move it:
> If this really is a board specific sample (which I don't think from the look of it),

So i'm aligned with @jfischer-no on his view to be optimal solution, although I do accept the `samples/boards` approach if code is board centric.

jfischer-no replied:
> I initially had it there but it was recommended that I move it to boards. I'm inclined to leave this sample here because I have another PR that implements the full USBC subsystem stack that included a more general sample.

I also assumed that it would be followed by an USBC stack. Nevertheless, you may be putting stones in your own way. A sample in `samples/subsys` can be revised/simplified when USBC stack is ready, sample in `samples/boards/stm32` must probably be removed.

jfischer-no commented on drivers/usbc/tcpc/Kconfig:
```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+config USBC_TCPC
```
Suggest `menuconfig USBC_TCPC` or `menuconfig USBC` somewhere, also it is better to add _DRIVER, `menuconfig USBC_DRIVER`, if there will be an USBC stack.

erwango replied:
This was my request to remove the `_DRIVER` which was initially there (for the reason we generally don't add _DRIVER). But if that's a practice used for potential disambiguation that's fine with me

jfischer-no replied:
> But if that's a practice used for potential disambiguation that's fine with me

I think there is a danger of confusion here, that is why we have for example Kconfig options `USB_DEVICE_DRIVER` and `USB_DEVICE_STACK`.

sambhurst replied:
I changed it to menuconfig USBC_TCPC_DRIVER

jfischer-no replied:
> I changed it to menuconfig USBC_TCPC_DRIVER

Actually not, did you push a wrong state?

jfischer-no commented on samples/boards/stm32/usbc/sink/prj.conf:
```diff
@@ -0,0 +1,5 @@
+CONFIG_USBC_TCPC=y
+CONFIG_USBC_TCPC_STM32=y
```
This is not necessary, because USBC_TCPC_STM32 is by default y if a compatible node is defined.

sambhurst replied:
Done

jfischer-no replied:
not addressed

tejlmand reviewed (dismissed):
lgtm

sambhurst commented on include/drivers/usbc/usbc_pd.h:
```diff
@@ -0,0 +1,843 @@
+/* usbc_pd.h - USBC Power Delivery */
```
Done

jfischer-no replied:
`/* usbc_pd.h - USBC Power Delivery */` is still there

sambhurst commented on include/drivers/usbc/usbc_pd.h:
```diff
@@ -0,0 +1,843 @@
+ *        See Table 6-75 Value Parameters
+ *        Parameter Name: MaxExtendedMsgLegacyLen
+ */
+#define PD_MAX_EXTENDED_MSG_LEGACY_LEN   26
```
Done

jfischer-no replied:
Unfortunately not changed

sambhurst replied:
I'm not sure what happened but I clearly uploaded the wrong files or I undid my changes somehow.

sambhurst commented on include/drivers/usbc/usbc_pd.h:
```diff
@@ -0,0 +1,843 @@
+		/** Maximum Voltage in 50mV units */
+		uint32_t max_voltage : 10;
+		/** Variable supply. SET TO PDO_VARIABLE  */
+		enum pdo_src_type type : 2;
```
Ack

sambhurst commented on include/drivers/usbc/usbc_tcpc.h:
```diff
@@ -0,0 +1,1067 @@
+/* usbc_tcpc.h - USB Type-C Port Controller driver interface */
```
Done

tejlmand reviewed (dismissed):
>

sambhurst commented:
Hello @jfischer-no, are there any other recommended changes you have?

VenkatKotakonda commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1463 @@
+	do {
+		ucpd_manage_tx(info);
+		/* Look at events only once. */
+		info->evt = 0;
```
evt variable getting updated in interrupt context and worker context(thread)
if this variable get updated in isr while worker in this loop, that updated flags will get cleared here.
my recommendation is to clear only the flags which are processed above in an atomic way.

sambhurst replied:
I should have spotted this. Thanks

sambhurst replied:
I should have noticed this. Thanks

VenkatKotakonda commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1463 @@
+	 * UCPD driver based on USB-PD rx messages. These 2 types of
+	 * transmit paths are managed via events.
+	 *
+	 * UCPD generated GoodCRC messages, are the priority path as
```
Does that mean the stm32 will auto ack the Rx messages with good crc without driver intervention?
if so ... make sense as the goodcrc timing requirement is ~1msec which is hard to achieve with driver code.

sambhurst replied:
The STM32 UCPD does not automatically send a GoodCRC when a message is received, so it must be done in the driver. From tests, I've seen the time vary from ~450 to 800uS.  The spec states that max time is 1100uS

VenkatKotakonda replied:
sending goodcrc from a work item is not a good idea in my opinion.
this work can be blocked for more than 1msec or for a long time if the previous work (by some other module/app)  did not complete its work with in this time.
Note: system work queue will run in sequence order in which the works are queued. Hence if previous work sleep/block for >1msec then the goodcrc which is expected to send within 1msec may not be sent within that time.
This piece of the code might be working now as no other task in your platform blocking work queue more than 1msec.

sambhurst replied:
I changed the code so that the GoodCRC is sent from the ISR instead of the work queue

VenkatKotakonda commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1463 @@
+	tx_header.extended = 0;
+
+	/* Good CRC is header with no other objects */
+	data->ucpd_tx_buffers[TX_MSG_GOOD_CRC].msg_len = 2;
```
number 2 can be replaced with a macro like USBC_CONTROL_MSG_LEN or size of header

sambhurst replied:
Done

VenkatKotakonda commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1463 @@
+	data->ucpd_tx_buffers[TX_MSG_GOOD_CRC].type = tx_type;
+
+	/* Notify ucpd task that a GoodCRC message tx request is pending */
+	info->evt |= UCPD_EVT_GOOD_CRC_REQ;
```
offloading good crc to worker will add undefined delay ( based on pending work item in the work queue).
how can the gooc-crc timing of ~1msec be achieved all the time?

sambhurst replied:
Same comment above

VenkatKotakonda commented on drivers/usbc/tcpc/ucpd_stm32.c:
```diff
@@ -0,0 +1,1463 @@
+	struct tcpc_data *data = info->dev->data;
+
+	if (info->evt & UCPD_EVT_EVENT_CC) {
+		info->handler(info->dev, info->data, TCPC_ALERT_CC_STATUS);
```
it is good practice to check for valid handler against NULL before calling.
as there are multiple places where handler is used, may adding new static function may help.

static void ucpd_nofity_handler(dev, data, status)
{
        if (dev->handler) {
                   dev->handler(dev, data, status);
        }
}

sambhurst replied:
Done

sambhurst commented:
Hello @jfischer-no, could you take another look at this PR at your earliest convenience? Thank you

jfischer-no commented on samples/boards/stm32/usbc/sink/src/usbc_snk.c:
```diff
@@ -0,0 +1,775 @@
+{
+	struct tc_t *tc = tc_obj;
+
+	printk("UnAttached.SNK\n");
```
Better to use either LOG_* or printk, since it looks like it is only there for logging purposes I suggest to use LOG_ instead of printk in `usbc_snk.c`

sambhurst replied:
I've changed the prinks to LOG_INF

tejlmand reviewed (approved):
approved.

I still think the optimal solution is to make the sample more generic, as discussed https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r811472924 and https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r796752627.
Although moving sample to `samples/boards` is sub-optimal, it is still an acceptable solution.
